qiskit/circuit/library/__init__.py

```diff
@@ -105,3 +113,4 @@
 )
 from .fourier_checking import FourierChecking
 from .quantum_volume import QuantumVolume
+from .diagonal import Diagonal
```

Comment:
Should we put this into a separate `generalized_gates` folder? Since the multi-controlled and uniformly-controlled gates will also follow.